### DISCUSSION TITLE
Page Dynamic Edit - Comportamento inesperado

### DISCUSSION THREAD

**Discussion by wsteixeira** [po-ui/po-angular#1126] [Tire suas dúvidas] [answered]
Estou usando  o código abaixo e o evento do "validate", não é disparado caso tente alterar diretamente as opções do "options", ele só é disparado após a primeira vez que a tecla "TAB" é acionada. Segue abaixo um print do codigo e um link para simulação do comportamento  [https://stackblitz.com/edit/po-ui-ym5kte?file=src/app/app.component.ts](https://stackblitz.com/edit/po-ui-ym5kte?file=src/app/app.component.ts).

```
  readonly fields: Array<PoDynamicFormField> = [
    { property: 'id', label: 'User ID', key: true },
    { property: 'genre', options: ['female', 'male', 'others'], validate: this.onChangeField.bind(this) },
  ]

  onChangeField(changeValue): PoDynamicFormFieldValidation {
    if (changeValue.property === 'genre') {
      alert ('Changed');
    }
    return {};
  }
```

**Comment by alinelariguet** [comment]
Boa tarde @wsteixeira,
Eu abri o link do Stackblitz e está sendo disparado o 'changed' toda vez que altero a opção selecionada. É isso mesmo?
Att.

**Comment by wsteixeira** [comment]
Boa noite @alinelariguet,

Para simular o problema você precisa clicar **diretamente** nas opções do "**Genre**", ou seja, **não** pode clicar no "**User Id**", caso clique primeiro no "**User Id**" e depois no "**Genre**" não vai conseguir simular o problema.

**Comment by wsteixeira** [comment, marked as answer]
Boa noite @alinelariguet, obrigado pelo retorno!

Sequencia para simulação:

1. **Não** clique no campo "**User ID**";
2. Clique diretamente em uma das opções do campo "Genre" e tecle **TAB**, observe que o "changed" não será disparado;
3. Clique novamente em uma das opções do campo "Genre", observe que o "changed" será disparado mesmo sem teclar o **TAB**;
